From 7f98433479e0e4087ad4b7b06cba00d604b7d1f1 Mon Sep 17 00:00:00 2001
From: Peter Tworek <tworaz666@gmail.com>
Date: Wed, 25 Mar 2015 22:16:48 +0100
Subject: [PATCH 2/7] Don't use EGL_KHR_surfaceless_context extension on X11
 with EGL.

For some reason this crashes inside mesa.

Upstream bug: https://code.google.com/p/chromium/issues/detail?id=462948
Mesa bug: https://bugs.freedesktop.org/show_bug.cgi?id=90821

Signed-off-by: Peter Tworek <tworaz666@gmail.com>
---
 ui/gl/gl_surface_egl.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ui/gl/gl_surface_egl.cc b/ui/gl/gl_surface_egl.cc
index f85258f..5c10350 100644
--- a/ui/gl/gl_surface_egl.cc
+++ b/ui/gl/gl_surface_egl.cc
@@ -335,6 +335,13 @@ bool GLSurfaceEGL::InitializeOneOff() {
   // Check if SurfacelessEGL is supported.
   g_egl_surfaceless_context_supported =
       HasEGLExtension("EGL_KHR_surfaceless_context");
+#if defined(USE_OZONE)
+  // Hopefully temporary workaround for:
+  // https://code.google.com/p/chromium/issues/detail?id=462948
+  if (!getenv("WAYLAND_DISPLAY")) {
+    g_egl_surfaceless_context_supported = false;
+  }
+#endif
   if (g_egl_surfaceless_context_supported) {
     // EGL_KHR_surfaceless_context is supported but ensure
     // GL_OES_surfaceless_context is also supported. We need a current context
-- 
2.4.10

